File: android-tv/app/src/main/java/com/homektv/tv/net/SnapshotRevisionPolicy.kt
package com.homektv.tv.net

/** Rejects legacy/older snapshots after a revisioned snapshot has been applied. */
internal object SnapshotRevisionPolicy {
    fun accepts(currentRevision: Long, incomingRevision: Long): Boolean =
        currentRevision <= 0L ||
            (incomingRevision > 0L && incomingRevision >= currentRevision)
}

File: windows-player/src/HomeKtv.Windows/Protocol/SnapshotRevisionPolicy.cs
namespace HomeKtv.Windows.Protocol;

/**
 * Accepts legacy revision 0 only until the first revisioned snapshot arrives;
 * afterwards an older or legacy response cannot roll the projection back.
 */
public static class SnapshotRevisionPolicy
{
    public static bool ShouldApply(long currentRevision, long incomingRevision) =>
        incomingRevision <= 0
            ? currentRevision <= 0
            : currentRevision <= 0 || incomingRevision >= currentRevision;
}
